packages/eventual-send/src/index.js

```diff
@@ -304,42 +381,52 @@ export function makeHandledPromise() {
       rejectHandled,
       resolveWithPresence,
     );
+
     return handledP;
   }
 
-  HandledPromise.prototype = Promise.prototype;
-  Object.setPrototypeOf(HandledPromise, Promise);
-
   function isFrozenPromiseThen(p) {
     return (
       isFrozen(p) &&
       getPrototypeOf(p) === Promise.prototype &&
       Promise.resolve(p) === p &&
-      gopd(p, 'then') === undefined
+      getOwnPropertyDescriptor(p, 'then') === undefined
     );
   }
 
+  /** @type {import('.').HandledPromiseStaticMethods} */
   const staticMethods = {
-    get(target, key) {
-      return handle(target, 'get', key);
+    get(target, prop) {
+      prop = coerceToObjectProperty(prop);
+      return handle(target, 'get', [prop]);
     },
-    getSendOnly(target, key) {
-      handle(target, 'get', key);
+    getSendOnly(target, prop) {
+      prop = coerceToObjectProperty(prop);
+      handle(target, 'getSendOnly', [prop]);
     },
     applyFunction(target, args) {
-      return handle(target, 'applyMethod', undefined, args);
+      // Ensure args is an array.
+      args = [...args];
+      return handle(target, 'applyFunction', [args]);
     },
     applyFunctionSendOnly(target, args) {
-      handle(target, 'applyMethod', undefined, args);
+      // Ensure args is an array.
+      args = [...args];
+      handle(target, 'applyFunctionSendOnly', [args]);
     },
-    applyMethod(target, key, args) {
-      return handle(target, 'applyMethod', key, args);
+    applyMethod(target, prop, args) {
+      prop = coerceToObjectProperty(prop);
+      // Ensure args is an array.
+      args = [...args];
+      return handle(target, 'applyMethod', [prop, args]);
     },
-    applyMethodSendOnly(target, key, args) {
-      handle(target, 'applyMethod', key, args);
+    applyMethodSendOnly(target, prop, args) {
+      prop = coerceToObjectProperty(prop);
+      // Ensure args is an array.
+      args = [...args];
+      handle(target, 'applyMethodSendOnly', [prop, args]);
     },
     resolve(value) {
-      ensureMaps();
       // Resolving a Presence returns the pre-registered handled promise.
       let resolvedPromise = presenceToPromise.get(value);
       if (!resolvedPromise) {
```

Comment:
Lemme see if I correctly understand the "Assimilate the thenable" code below. The key is that it can only be invoked if `value` is a promise but it does not pass the `isFrozenPromiseThen` test. Only in that case to we get there. The point of that code is to treat `value` as if it is a non-promise thenable. I know I knew this once and it still took me awhile. Worth a comment.